drivers/pwm/pwm_sf32lb.c

```diff
@@ -0,0 +1,167 @@
+/*
+ * Copyright (c) 2025, Qingsong Gou <gouqs@hotmail.com>
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT sifli_sf32lb_pwm
+
+#include <zephyr/kernel.h>
+#include <zephyr/drivers/pwm.h>
+#include <zephyr/drivers/clock_control/sf32lb.h>
+#include <zephyr/drivers/pinctrl.h>
+#include <zephyr/logging/log.h>
+#include <zephyr/irq.h>
+
+#include <register.h>
+
+#include <bf0_hal.h>
+#include <bf0_hal_tim.h>
+
+#define GPT_CR1   offsetof(GPT_TypeDef, CR1)
+#define GPT_PSC   offsetof(GPT_TypeDef, PSC)
+#define GPT_ARR   offsetof(GPT_TypeDef, ARR)
+#define GPT_CCR1  offsetof(GPT_TypeDef, CCR1)
+#define GPT_CCER  offsetof(GPT_TypeDef, CCER)
+#define GPT_CCMR1 offsetof(GPT_TypeDef, CCMR1)
+#define GPT_EGR   offsetof(GPT_TypeDef, EGR)
+
+#define GPT_CCMRX(ch) (GPT_CCMR1 + (ch >> 1U))
+#define CCRX(ch)      (GPT_CCR1 + ((ch) << 2U))
+
+LOG_MODULE_REGISTER(pwm_sf32lb, CONFIG_PWM_LOG_LEVEL);
+
+struct pwm_sf32lb_config {
+	const struct pinctrl_dev_config *pcfg;
+	const struct device *clock_dev;
+	uint16_t clock_id;
```

Comment:
there are common facilities to grab sf32 clocks

Reply:
```
/**
 * @brief Initialize a `sf32lb_clock_dt_spec` structure from a parent DT instance.
 *
 * @param index DT instance index
 */
#define SF32LB_CLOCK_DT_INST_PARENT_SPEC_GET(index)                                                \
	{                                                                                          \
		.dev = DEVICE_DT_GET(DT_CLOCKS_CTLR(DT_INST_PARENT(index))),                       \
		id = DT_CLOCKS_CELL(DT_INST_PARENT(index), id),                                    \
	}
```
this style?

Reply:
yes